optimizer/plan/planbuilder_join.go

```diff
@@ -584,7 +584,9 @@ func (b *planBuilder) buildJoin(sel *ast.SelectStmt) Plan {
 	whereConditions := splitWhere(sel.Where)
 	for _, whereCond := range whereConditions {
 		if !path.attachCondition(whereCond, nil) {
-			log.Errorf("Failed to attach where condition.")
+			// TODO: Find a better way to handle this condition.
+			path.conditions = append(path.conditions, whereCond)
```

Comment:
"select c1 from t1 where c1 in (select c2 from t2 where t2.c1=t1.c1)"
This is a temp solution. cooocood will fix it after this pr.